File: web3rpc/rpc-specs/code-samples/python/debug/blockchainInspection/getBlockRlp.py
from web3 import Web3
from web3py_ext import extend

host = "https://public-en-kairos.node.kaia.io"

blockHash = "latest"

w3 = Web3(Web3.HTTPProvider(host))
debug_response = w3.debug.get_block_rlp(blockHash)

print(debug_response)

File: web3rpc/rpc-specs/code-samples/javascript/debug/blockchainInspection/getBlockRlp.js
const { JsonRpcProvider } = require("@kaiachain/ethers-ext");
(() => {
  const provider = new JsonRpcProvider("https://public-en-kairos.node.kaia.io");
  const blockNumber = "200";

  provider.debug
    .getBlockRlp(blockNumber, {}, (err, data, response) => {})
    .then((data) => {
      console.log(data);
    });
})();
